lib/SILOptimizer/Utils/CanonicalizeOSSALifetime.cpp

```diff
@@ -781,6 +899,11 @@ bool CanonicalizeOSSALifetime::canonicalizeValueLifetime(SILValue def) {
   // Step 2: compute boundary
   PrunedLivenessBoundary boundary;
   findExtendedBoundary(boundary);
+  if (ononeMode) {
+    extendUnconsumedLiveness(boundary);
+    boundary.clear();
+    findExtendedBoundary(boundary);
```

Comment:
It looks like `findExtendedBoundary` supposed to initialize a new boundary? If so, please assert that the boundary is empty inside that utility.

Reply:
That's right.  An assertion would be nice, I agree.  Added.